### PR TITLE
feat: load and validate workflows (#2)

### LINKED ISSUE #2: Load and validate Markdown workflows
## Goal
Factory discovers versioned Markdown workflow files and validates the prompt and trigger contract before execution.

## Background
See [docs/design.md](https://github.com/owainlewis/factory/blob/main/docs/design.md). A workflow is `.factory/workflows/<id>.md`; its filename is the workflow ID, TOML frontmatter declares either a cron schedule or ticket label, and the Markdown body is the agent prompt.

## Acceptance criteria
- Factory discovers regular Markdown files under each configured repository's `.factory/workflows/` directory.
- The filename becomes a stable lowercase workflow ID.
- TOML frontmatter supports exactly one V1 trigger: `schedule` with timezone, or `label`.
- Optional runtime and timeout values override global defaults.
- Missing prompts, duplicate IDs, unknown fields, invalid cron/timezone values, invalid labels, unsafe files, and simultaneous schedule and label triggers produce actionable validation errors.
- One invalid workflow does not hide validation results for other workflows.
- `factory workflows` lists resolved repository, workflow ID, trigger, runtime, timeout, and validity.
- Loading workflows never runs their prompts.

## Verify
- Run `cargo test` with fixtures covering valid scheduled and label workflows plus every validation failure above.
- Run `factory workflows --config <fixture>` and inspect the resolved catalog.
- Run the standard format and Clippy checks.

## Depends on
- #1

### PR DESCRIPTION
**Pull request by owainlewis**
## Summary

- discover Markdown workflows from each configured repository and infer stable lowercase IDs
- parse strict TOML frontmatter, resolve runtime and timeout defaults, and validate cron or label triggers
- add a read-only workflows catalog command with aggregate actionable errors
- contain workflow discovery to configured repositories, read prompts through no-follow handles, and escape terminal control characters

Closes #2

## Acceptance proof

- valid scheduled and label workflows resolve repository, ID, trigger, runtime, timeout, and prompt
- missing prompts, duplicate IDs, unknown fields, invalid cron/timezone/labels, unsafe files, and multiple triggers have fixture coverage
- invalid workflows remain visible beside valid workflows and make the command return non-zero
- prompt loading is verified to have no execution side effects
- symlinked ancestors, path-swap symlinks, and terminal control injection have regression coverage

## Checks

- cargo fmt --all -- --check
- cargo test (26 passed)
- cargo clippy --all-targets --all-features -- -D warnings
- git diff --check

## Review

Fresh local and GitHub reviews found filesystem and terminal-safety edge cases. The findings were fixed, regression-tested, replied to, and resolved before handoff.

### CODE REVIEW

File: src/workflow.rs [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+                timezone,
+            } => write!(formatter, "schedule {expression:?} ({timezone})"),
+            Self::Label(label) => write!(formatter, "label {label:?}"),
+        }
+    }
+}
+
+fn load_repository(repository: &Path, config: &Config) -> Vec<WorkflowEntry> {
+    let directory = repository.join(WORKFLOW_DIRECTORY);
+    let metadata = match fs::symlink_metadata(&directory) {
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply src/workflow.rs]: Fixed in fe43258. Workflow discovery now canonicalizes the resolved workflows directory and rejects it when it escapes the configured canonical repository. Each regular workflow file is also canonicalized and checked before reading. Added a regression test for `.factory` symlinked to an external directory, then reran fmt, all 24 tests, strict Clippy, and diff checks.

**Comment by owainlewis** [comment]
@codex review

File: src/workflow.rs [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+                entry.repository.display(),
+                entry.id,
+                trigger,
+                runtime,
+                timeout,
+                validity
+            )?;
+        }
+        Ok(())
+    }
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply src/workflow.rs:117]: Fixed in ac2601f. Every header and dynamic catalog cell now passes through control-character escaping before tabular rendering. Added a regression with ESC bytes in the repository path, workflow ID, and runtime; the rendered output contains escaped text and no raw ESC.

File: src/workflow.rs [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+    paths.sort();
+
+    entries.extend(
+        paths
+            .into_iter()
+            .filter(|path| is_markdown_path(path))
+            .map(|path| load_file(repository, &path, config)),
+    );
+    entries
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply src/workflow.rs:233]: Fixed in ac2601f. On Unix, workflow files are opened with O_NOFOLLOW and read from that same descriptor after its metadata confirms a regular file. The loader no longer reopens the checked path with read_to_string. Added a direct regression proving the no-follow open rejects a symlink.

**Comment by owainlewis** [comment]
@codex review
